cluster_management/src/main/java/com/pinterest/rocksplicator/Participant.java

```diff
@@ -204,12 +215,43 @@ private Participant(String zkConnectString, String clusterName, String instanceN
           port, zkConnectString, clusterName, useS3Backup, s3BucketName);
     } else if (stateModelType.equals("Bootstrap")) {
       stateModelFactory = new BootstrapStateModelFactory(instanceName.split("_")[0],
-              port, zkConnectString, clusterName);
+          port, zkConnectString, clusterName);
+    } else if (stateModelType.equals("MasterSlave;Task")) {
+      stateModelType = "MasterSlave";
+      stateModelFactory = new MasterSlaveStateModelFactory(instanceName.split("_")[0],
+          port, zkConnectString, clusterName, useS3Backup, s3BucketName);
+
+      Map<String, TaskFactory> taskFactoryRegistry = new HashMap<>();
+      // TODO: register backup/restore factories
+      // taskFactoryRegistry.put("Backup", new BackupTaskFactory());
+      // taskFactoryRegistry.put("Restore", new RestoreTaskFactory());
+
+      TaskStateModelFactory helixTaskFactory =
+          new TaskStateModelFactory(helixManager, taskFactoryRegistry);
+      stateMach.registerStateModelFactory("Task", helixTaskFactory);
```

Comment:
Can we just create and fill taskFactoryRegistry here and move registerStateModelFactory "Task" to be after? This will help remove code duplication.